app/javascript/packages/session/session-timeout-modal-element.ts

```diff
@@ -0,0 +1,100 @@
+import { forceRedirect } from '@18f/identity-url';
+import type { ModalElement } from '@18f/identity-modal';
+import type { CountdownElement } from '@18f/identity-countdown/countdown-element';
+import { requestSessionStatus, extendSession } from './requests';
+
+class SessionTimeoutModalElement extends HTMLElement {
+  statusCheckTimeout: number;
+
+  connectedCallback() {
+    this.bindButtonEvents();
+    this.scheduleStatusCheck({ timeout: this.timeout });
+  }
+
+  disconnectedCallback() {
+    window.clearTimeout(this.statusCheckTimeout);
+  }
+
+  /**
+   * Amount of time before timeout to show modal, in milliseconds.
+   */
+  get warningOffset(): number {
+    return Number(this.getAttribute('warning-offset-in-seconds')!) * 1000;
+  }
+
+  get timeout(): Date {
+    return new Date(this.getAttribute('timeout')!);
+  }
+
+  get timeoutURL(): string {
+    return this.getAttribute('timeout-url')!;
+  }
+
+  get modal(): ModalElement {
+    return this.querySelector('lg-modal')!;
+  }
+
+  get keepAliveButton(): HTMLButtonElement {
+    return this.querySelector<HTMLButtonElement>('.lg-session-timeout-modal__keep-alive-button')!;
+  }
+
+  get signOutButton(): HTMLButtonElement {
+    return this.querySelector<HTMLButtonElement>('.lg-session-timeout-modal__sign-out-button')!;
+  }
+
+  get countdownElements(): CountdownElement[] {
+    return Array.from(this.querySelectorAll('lg-countdown'));
+  }
+
+  bindButtonEvents() {
+    this.keepAliveButton.addEventListener('click', () => this.keepAlive());
+  }
+
+  keepAlive() {
+    this.modal.hide();
+    this.countdownElements.forEach((countdown) => countdown.stop());
+    extendSession();
+  }
+
+  async checkStatus() {
+    const { isLive, timeout } = await requestSessionStatus();
+
+    if (isLive) {
+      const timeRemaining = timeout.valueOf() - Date.now();
+      const showWarning = timeRemaining < this.warningOffset;
```

Comment:
what if we named all of these with units? then we could simplify the comment on `warningOffset` as well?
```suggestion
      const millisecondsRemaining = timeout.valueOf() - Date.now();
      const showWarning = millisecondsRemaining < this.warningOffsetMilliseconds;
```

Reply:
Yeah, I suppose we should do that for added clarity. I think I have some internal reluctance since JavaScript measures all time durations in miliseconds, so the unit of time is a bit more redundant in that sense, vs. in Rails where we can easily convert between different units of time.

Aside from clarity, it might also be worth looking to the future where those assumptions may no longer be accurate with the proposed [`Temporal.Duration`](https://tc39.es/proposal-temporal/docs/#Temporal-Duration).

Reply:
Updated naming in 9a15272.